src/idpLoader.ts

```diff
@@ -0,0 +1,91 @@
+import * as https from "https";
+import { DOMParser } from "xmldom";
+import { log } from "./utils/logger";
+
+interface IDPMetadata {
+  cert: ReadonlyArray<string>;
+  entityID: string;
+  entryPoint: string;
+  logoutUrl: string;
+}
+
+const EntityDescriptorTAG = "md:EntityDescriptor";
+const X509CertificateTAG = "ds:X509Certificate";
+const SingleSignOnServiceTAG = "md:SingleSignOnService";
+const SingleLogoutServiceTAG = "md:SingleLogoutService";
+
+export async function parseIdpMetadata(): Promise<ReadonlyArray<IDPMetadata>> {
+  const IDP_METADATA_ULR =
+    process.env.IDP_METADATA_ULR ||
+    "https://registry.spid.gov.it/metadata/idp/spid-entities-idps.xml"; // TODO: remove default configuration?
+  const ipdMetadataPage = await getRequest(IDP_METADATA_ULR); // TODO: Failure on download remote configuration will throw an exception
+  const domParser = new DOMParser().parseFromString(ipdMetadataPage);
+  const entityDescriptors = domParser.getElementsByTagName(EntityDescriptorTAG);
+  return Array.from(entityDescriptors).reduce(
+    (idp: ReadonlyArray<IDPMetadata>, element: Element) => {
+      if (validateEntityDescriptorFormat(element)) {
+        const elementInfo = {
```

Comment:
you should not cast to `string`, you should use io-ts to decode the object to a valid `elementInfo`, i.e.:

given that you defined an io-ts type for `ElementInfo` you can:

```ts
const elementInfoOrErrors = ElementInfo.decode({
  entityID: element.getAttribute("entityID"),
  // ...
})
```

Reply:
fix in commit 420b3cc